api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java

```diff
@@ -294,4 +300,60 @@ public String getTestComplexObsFilePath() {
 		}
 		return obsList;
 	}
+	
+	/**
+	 * Boilerplate method to construct and save a complex obs with random data.
+	 * 
+	 * @param patient the patient
+	 * @param concept the associated concept
+	 * @return obs
+	 */
+	public Obs saveComplexObs(Patient patient, Concept concept) throws IOException {
+		Obs obs = new Obs();
+		byte[] randomData = new byte[20];
+		obs.setConcept(concept);
+		obs.setObsDatetime(new Date());
+		obs.setPerson(patient);
+		
+		new Random().nextBytes(randomData);
+		
+		String filename = RandomStringUtils.randomAlphabetic(7) + ".ext";
+		MockMultipartFile multipartRandomFile = new MockMultipartFile(FilenameUtils.getBaseName(filename), filename,
+		        "application/octet-stream", randomData);
+		obs.setComplexData(
+		    complexDataHelper.build(ValueComplex.INSTRUCTIONS_DEFAULT, multipartRandomFile.getOriginalFilename(),
+		        multipartRandomFile.getBytes(), multipartRandomFile.getContentType()).asComplexData());
+		
+		return context.getObsService().saveObs(obs, null);
+	}
```

Comment:
Why didn't you make [this](https://github.com/openmrs/openmrs-module-attachments/blob/bd8f77ba063203a6898b33ef4a856c64444b9738/api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java#L259) parameterizable with a `concept` argument?

Reply:
Updated